components/script/dom/document.rs

```diff
@@ -3459,6 +3462,97 @@ impl Document {
         // implement the Permissions Policy specification.
         true
     }
+
+    /// <https://w3c.github.io/IntersectionObserver/#dom-intersectionobserver-intersectionobserver>
```

Comment:
Here I don't think the link is useful, because it does not link to anything describing "adding an intersection observer". I think it would be better to point out that this does not follow the spec as per https://github.com/w3c/IntersectionObserver/issues/525

(same below)